lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -206,6 +211,50 @@ protected function _getPortableTableColumnDefinition($tableColumn)
         return $column;
     }
 
+
+    /**
+     * Return Doctrine/Mysql-compatible column default values for MariaDB 10.2.7+ servers.
+     *
+     * - Since MariaDb 10.2.7 column defaults stored in information_schema are now quoted
+     *   to distinguish them from expressions (see MDEV-10134 for a what is an expression).
+     * - CURRENT_TIMESTAMP, CURRENT_TIME, CURRENT_DATE are stored in information_schema
+     *   as current_timestamp(), currdate(), currtime()
+     * - Literal escaping is normalized in information schema (store "''" instead of "\'")
+     * - Note: columnDefault should only be null when column is not_nullable otherwise
+     *   a string 'NULL' should be given. Unfortunately, it's not 100% correct in case of
+     *   upgrade from previous versions: see https://jira.mariadb.org/browse/MDEV-14053.
+     *   Otherwise we could simplify the condition === 'NULL' || === null by adding the parameter
+     *   is_nullable.
+     *
+     * @link https://mariadb.com/kb/en/library/information-schema-columns-table/
+     * @link https://jira.mariadb.org/browse/MDEV-13132
+     * @link https://jira.mariadb.org/browse/MDEV-14053
+     *
+     * @param null|string $columnDefault default value as stored in information_schema for MariaDB >= 10.2.7
+     */
+    private function getMariaDb1027ColumnDefault(MariaDb102Platform $platform, ?string $columnDefault) : ?string {
```

Comment:
The method name is inconsistent with `MariaDb102Platform`.

Reply:
> I finally chose 'MariaDB102Platform' instead of 'MariaDB1027Platform'. GA started at 10.2.6 but information schema (breaking) changes happened in 10.2.7. In current P/R <= 10.2.6 will be considered as MySQLPlatform and works. I don't think it's a big deal, if you think I'll update accordingly.

I'll update to MariaDB1027Platform. 

side note: as a first time contributor: many reviews, a lot of code fix. Was great, I learned a lot and very helpful persons ! Unfortunatly, architectural or naming questions I asked found less responses... So I fall back to existing code to choose by myself: MySQL57 is actually 5.7.9.  Still the method name must be 1027. All cool, but just to let you know some things that can keep popping up. I'll summarize my remaining questions in a couple of hours, good if I can help me to find answer.